UnityProject/Assets/Scripts/Items/Cargo/Wrapping/WrappedBase.cs

```diff
@@ -24,25 +26,23 @@ public abstract class WrappedBase: MonoBehaviour
 		[SerializeField] [Tooltip("Time to unwrap.")]
 		private float timeToUnwrap = 2;
 
-		private PushPull content;
-		private PushPull pushPull;
 		private Attributes attributes;
 		protected SpriteHandler spriteHandler;
 
-		protected virtual void Awake()
+		protected virtual void OnEnable()
 		{
-			pushPull = GetComponent<PushPull>();
 			spriteHandler = GetComponentInChildren<SpriteHandler>();
 			attributes = GetComponent<Attributes>();
 		}
 
 		public void SetContent(GameObject toWrap)
 		{
-			content = toWrap.GetComponent<PushPull>();
+			StoreObject(toWrap);
 			var exportCost = toWrap.GetComponent<Attributes>().ExportCost;
 			UpdateExportCost(exportCost);
-			content.parentContainer = pushPull;
-			content.VisibleState = false;
+			toWrap.GetComponent<PushPull>().VisibleState = false;
+			toWrap.GetComponent<ObjectBehaviour>().parentContainer = GetComponent<ObjectBehaviour>();
```

Comment:
I see you're using the two here too. We should really drop one of these at some point.

Reply:
Fair enough, replaced ObjectBehavior in this instance with PushPull